modules/messages/src/inbound_lane.rs

```diff
@@ -545,4 +546,29 @@ mod tests {
 			);
 		});
 	}
+
+	#[test]
+	fn no_off_by_one_when_first_message_of_single_relayer_is_confirmed() {
```

Comment:
Nit: I would rename this to something like `first_message_of_single_relayer_is_confirmed_correctly`